js/directives/lucca/user-picker.js

```diff
@@ -0,0 +1,322 @@
+											hasHomonyms = true;
+										}
+									}
+								});
+							});
+			return hasHomonyms;
+		};
+
+		var getHomonymsProperties = function () {
+			var homonyms = _.where($scope.users, { hasHomonyms: true });
+			var homonymsArray = [];
+
+			loadPropertiesFromUrls(homonyms).then(function (response) {
+								// Add homonyms properties for each user
+								_.each(homonyms, function (user, item) {
+									// Get the returned user
+									var userWithProps = response[item].data.data.items[0];
+
+									// Add each property to the user
+									_.each($scope.properties, function (prop) {
+										var newProp = prop.split('.')[0];
+										user[newProp] = userWithProps[newProp];
+									});
+								});
+							});
+		};
+
+		var loadPropertiesFromUrls = function (homonyms) {
+			var urlCalls = [];
+			var queries = [];
+			var templateQuery = "/api/v3/users?id=";
+			var fields = "&fields=id,name,firstname,lastname,dtcontractend";
+			var deferred = $q.defer();
+
+			// WARNING: Do not check if the properties exist!
+			// WARNING: If they do not exist, the request will fail
+			_.each($scope.properties, function (prop) {
+				fields += "," + prop;
+			});
+
+			_.each(homonyms, function (user) {
+				queries.push(templateQuery + user.id + fields);
+			});
+
+			_.each(queries, function (query) {
+				urlCalls.push($http.get(query));
+			});
+
+			$q.all(urlCalls)
+			.then(
+				function (response) {
+					deferred.resolve(response);
+				},
+				function (error) {
+					deferred.reject(error);
+				}
+				);
+			return deferred.promise;
+		};
+
+		var countTimeout = function (input, delay) {
+			delay = delay || 5000; // default delay value 1s
+
+			// cancel previous timeout
+			if (timeout.count) {
+				console.log("-- cancel previous timeout");
+				$timeout.cancel(timeout.count);
+			}
+
+			// launch new timeout to save in _delay_ 
+			timeout.count = $timeout(function () {
+				doCount(input);
+			}, delay);
+		};
+
+		var doCount = function (input) {
+			var query = "/api/v3/users?name=like," + input + "&fields=collection.count"; // query for count
+
+			delete timeout.count;
+
+			$http.get(query)
+			.then(function (response) {
+					$scope.count = response.data.data.count; // To update with right query
+					_.last($scope.users).overflow = MAX_COUNT + "/" + $scope.count;
+				});
+		};
+
+		$scope.getProperty = function (user, prop) {
+			var propList = prop.split('.');
+			var value = user[_.first(propList)];
+
+			_.each(propList, function (item) {
+				if (value && (item !== _.first(propList))) {
+					value = value[item];
+				}
+			});
+
+			return value;
+		};
+
+		// Used by UserPickerMultiple
+		// Function executed when onSelect is fired
+		$scope.addSelectedUser = function () {
+			$scope.onSelect();
+			selectedUsersCount++;
+			// Update overflow message
+			if ($scope.count > MAX_COUNT) {
+				_.last($scope.users).overflow = (MAX_COUNT - selectedUsersCount) + "/" + $scope.count;
+			}
+		};
```

Comment:
in multi select you ask for `MAX_COUNT +1 +number of already selected users` users to be sure to always have `MAX_COUNT` results to display, right.

Now when i select one of the choice you may only have `MAX_COUNT-1` remaining available choices which is not okay, you should always have `MAX_COUNT` available choices so you may have to refresh the choices every time a user is selcted 

Also how do you handle pagination? lets take an example:

 - `MAX_COUNT = 2`
 - I select `alex`, `arthur` and `beatrix`. 
 - I type `a`, the directives ask for the `2+1+3=6` first users corresponding to `clue=a`
 - the server gives me `abigail`, `alex`, `alexis`, `alexia`, `arnaud`, `arthur`
 - the directives knows there are more results than what she will display but not yet how much
 - the directive displays `abigail`, `alexis`, `results 1-2 of ...`
 - the directive asks for the number of users corresponding to `clue=a`
 - the server answers 7. what does the directive display?
 - how many of those 7 have i already selected? how can you know what logic the server implements?

This shows that in case of a multi select if you want to handle pagination you must ask for __ALL__ the users verifying `clue=a`.

I am willing to scratch the async handling of pagination in the first draft. always ask for __ALL__ the users even in a normal select and compute the pagination label with the result. Let's see what kind of impact it has on the db server and web server and if it is too much optimise it when possible with the async pagination management we have now with the function `count` and `doCount`